src/cuttlefish_conf.erl

```diff
@@ -132,6 +132,22 @@ generate_element(MappingRecord) ->
             Comments = generate_comments(MappingRecord),
             Comments ++ [lists:flatten([ Field, " = ", cuttlefish_datatypes:to_string(Default, Datatype) ]), ""]
     end.
+
+get_default(MappingRecord) ->
+    %% Normally we use `default` to determine what value to use when generating
+    %% a config file, but `new_conf_value` can override that. The reason we need
+    %% a separate attribute to override `default` (instead of just changing the
+    %% default directly) is that `default` also affects default values used for
+    %% config keys that haven't been set to any particular value in the .conf file.
+    %% (See `cuttlefish_generator:add_defaults` for the relevant bits of code.)
+    Default = cuttlefish_mapping:default(MappingRecord),
+    NewConfValue = cuttlefish_mapping:new_conf_value(MappingRecord),
+
+    case NewConfValue of
+        undefined -> Default;
+        _ -> NewConfValue
+    end.
+
```

Comment:
Ah thanks, good call. I inherited the old code structure from the place I factored this out of, but your way is more succinct. I'll add that change in one moment.